fluent-package/templates/package-scripts/fluent-package/deb/postinst

```diff
@@ -145,37 +145,36 @@ migration_from_v4_post_process() {
     fi
 }
 
-local_plugin_migration=/tmp/<%= package_dir %>/.local_plugin_migration
 local_base_plugins=/tmp/<%= package_dir %>/.local_base_plugins
-local_missing_plugins=/tmp/<%= package_dir %>/.local_missing_plugins
 install_missing_plugins() {
-    if [ -f $local_plugin_migration ]; then
-        echo "Checking network connectivity..."
-	if ! curl --fail --silent -O https://rubygems.org/specs.4.8.gz; then
-	    echo "Can't install missing plugins automatically, please install the following plugins manually."
-	    cat $local_plugin_migration
-	else
-	    echo "Install missing plugins automatically..."
-	    /usr/sbin/fluent-gem list | grep fluent-plugin | cut -d' ' -f1 > $local_base_plugins
-	    # NOTE: Do not install already bundled plugins
-	    diff -u $local_base_plugins $local_plugin_migration | grep '^+fluent' | sed -e 's/\+//' > $local_missing_plugins
-	    if [ -s $local_missing_plugins ]; then
-		cat $local_missing_plugins | xargs sudo /usr/sbin/fluent-gem install --no-document
-	    else
-		echo "No need to install missing plugins..."
-	    fi
-	    rm -f $local_base_plugins
-	    rm -f $local_missing_plugins
-	fi
-	rm -f $local_plugin_migration
+    # Install missing gems (even though systemd is not available, it works)
+    if ! curl --fail --silent -O https://rubygems.org/specs.4.8.gz; then
+        echo "No network connectivity..."
+    else
+        if [ -f $local_base_plugins ]; then
+            local_current_plugins=$(/usr/sbin/fluent-gem list '^fluent-plugin-' --no-versions --no-verbose)
+            local_missing_plugins=$(grep --fixed-strings --line-regexp --invert-match "$local_current_plugins" $local_base_plugins) || true
+            if [ -n "$local_missing_plugins" ]; then
+                echo "Try to install the following plugins:"
+                echo $local_missing_plugins
+                for gem in $local_missing_plugins; do
+                    if ! /usr/sbin/fluent-gem install --no-document $gem; then
+                        echo "Can't install missing plugin automatically: please install $gem manually."
+                    fi
+                done
+            fi
```

Comment:
How about this?

* Use `|| true` in the sub-shell
* Don't use `-n` because `local_missing_plugins` can include an empty line due to `true` output.
  * `for` would automatically parse the words and ignore an empty line.

```suggestion
            local_missing_plugins=$(grep --fixed-strings --line-regexp --invert-match "$local_current_plugins" $local_base_plugins || true)
            for gem in $local_missing_plugins; do
                if ! /usr/sbin/fluent-gem install --no-document $gem; then
                    echo "Can't install missing plugin automatically: please install $gem manually."
                fi
                echo "$gem installed."
            done
```